junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -141,4 +141,134 @@ private static boolean hasParameters(Method method) {
 		}
 	}
 
+	/**
+	 * {@code DisplayNameGenerator} that supports {@code ReplaceUnderscores} to generate complete sentences.
+	 *
+	 * <p>This generator extends the functionality of {@link ReplaceUnderscores} by
+	 * generating a human-readable display names that form complete sentences divided each
+	 * class, nested class and test by a ({@code ','}).
+	 */
+	class IndicativeSentencesGenerator extends ReplaceUnderscores {
+
+		@Override
+		public String generateDisplayNameForClass(Class<?> testClass) {
+			String indicativeName = removeRootCharacter(super.generateDisplayNameForClass(testClass));
+			return indicativeName;
+		}
+
+		@Override
+		public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+			String indicativeName = classReplaceToIndicativeSentence(nestedClass);
+			return indicativeName;
+		}
+
+		@Override
+		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+			String methodName = underScoreNameFormatting(super.generateDisplayNameForMethod(testClass, testMethod));
+			String indicativeName = classReplaceToIndicativeSentence(testClass);
+			return indicativeName + ", " + methodName;
+		}
+
+		private String classReplaceToIndicativeSentence(Class<?> testClass) {
+			String classIndicativeSentence = null;
+			Class<?> classWithEnclosingParent = testClass.getEnclosingClass();
+			DisplayName classWithDisplayName = testClass.getAnnotation(DisplayName.class);
+			DisplayNameGeneration classWithAnnotation = testClass.getAnnotation(DisplayNameGeneration.class);
+
+			if (classWithEnclosingParent == null) {
+				if (classWithDisplayName != null)
+					classIndicativeSentence = classWithDisplayName.value();
```

Comment:
You should return the string here instead of storing it and have the return statement at the end of the method (same for the other occurences below). This way, you can also get rid of the initialization at line 173.

Reply:
I see, as I had the idea to concatenate with the parents recursively, so I decided that way, but yeah sure with the returns maybe a cleaner way.